specifications/api/swagger.yaml

```diff
@@ -1674,6 +1674,14 @@ x-maximumTokenBundleSize: &maximumTokenBundleSize
       enum:
         - byte
 
+x-transactionScriptValidity: &transactionScriptValidity
+  description: |
+    For Alonzo-era and later transactions, indicates whether the phase-2
+    monetary policy script (e.g. Plutus script) used in the transaction
+    validated or not. Previous to Alonzo-era, this is null.
+  nullable: true
+  type: boolean
+  default: null
```

Comment:
Hi @sevanspowell

I think it might be better if this was an _enumeration_, rather than a Boolean.

For example:
```yaml
  x-txScriptValidity: &txScriptValidity
    type: string
    enum:
      - unsupported
      - valid
      - invalid
```

I think this has the following advantages:

1. We can generate the `ToJSON` and `FromJSON` instances generically, rather than having to write them manually (which requires maintenance).
2. We don't need to deal with the missing case, as it's represented explicitly.

Thoughts?

Reply:
Yeah, I think that's great. Implementing now...

Reply:
1531fc7